doctr/datasets/wildreceipt.py

```diff
@@ -0,0 +1,141 @@
+        train: whether the subset should be the training one
+        use_polygons: whether polygons should be considered as rotated bounding box (instead of straight ones)
+        recognition_task: whether the dataset should be used for recognition task
+        **kwargs: keyword arguments from `AbstractDataset`.
+    """
+
+    def __init__(
+            self,
+            img_folder: str,
+            label_path: str,
+            train: bool = True,
+            use_polygons: bool = False,
+            recognition_task: bool = False,
+            **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            img_folder, pre_transforms=convert_target_to_relative if not recognition_task else None, **kwargs
+        )
+        # File existence check
+        if not os.path.exists(label_path) or not os.path.exists(img_folder):
+            raise FileNotFoundError(f"unable to locate {label_path if not os.path.exists(label_path) else img_folder}")
+
+        tmp_root = img_folder
+        self.train = train
+        np_dtype = np.float32
+        self.data: List[Tuple[Union[str, Path, np.ndarray], Union[str, Dict[str, Any]]]] = []
+
+        # define folder to write IMGUR5K recognition dataset
+        reco_folder_name = "WILDRECEIPT_recognition_train" if self.train else "WILDRECEIPT_recognition_test"
+        reco_folder_name = "Poly_" + reco_folder_name if use_polygons else reco_folder_name
+        reco_folder_path = os.path.join(os.path.dirname(self.root), reco_folder_name)
+        reco_images_counter = 0
+
+        if recognition_task and os.path.isdir(reco_folder_path):
+            self._read_from_folder(reco_folder_path)
+            return
+        elif recognition_task and not os.path.isdir(reco_folder_path):
+            os.makedirs(reco_folder_path, exist_ok=False)
+
+        with open(label_path, 'r') as file:
+            data = file.read()
+        # Split the text file into separate JSON strings
+        json_strings = data.strip().split('\n')
+        box: Union[List[float], np.ndarray]
+        _targets = []
+        for json_string in json_strings:
+            json_data = json.loads(json_string)
+            img_path = json_data['file_name']
+            annotations = json_data['annotations']
+            for annotation in annotations:
+                coordinates = annotation['box']
+                if use_polygons:
+                    # (x, y) coordinates of top left, top right, bottom right, bottom left corners
+                    box = np.array(
+                        [
+                            [coordinates[0], coordinates[1]],
+                            [coordinates[2], coordinates[3]],
+                            [coordinates[4], coordinates[5]],
+                            [coordinates[6], coordinates[7]],
+                        ],
+                        dtype=np_dtype
+                    )
+                else:
+                    box = self._convert_xmin_ymin(coordinates)
```

Comment:
No need to write an own function you can use the functions from `doctr.utils`

```
from .utils import polygon_to_bbox
box_targets = polygon_to_bbox(tuple((coordniates[i], coordinates[i + 1]) for i in range(0, len(coordinates), 2)))
box = [coord for coords in box_targets for coord in coords]
```
OR

write the logic directly here (function is only used onces) 
```
x, y = box[::2], box[1::2]
box = [min(x), min(y), max(x), max(y)]
```

I would prefer the sec way
:)

Reply:
@HamzaGbada can we use the secound suggestion please after reading this again i really don't like it 😅

Reply:
OK